frontend/src/scenes/session-recordings/components/replayActiveUsersTableLogic.ts

```diff
@@ -23,18 +23,47 @@ export const replayActiveUsersTableLogic = kea<replayActiveUsersTableLogicType>(
         countedUsers: {
             loadCountedUsers: async (_, breakpoint): Promise<{ person: PersonType; count: number }[]> => {
                 const q = hogql`
-                    select p, any (pp), count () as c
-                    from (
-                        select any (person_id) as p, any (person.properties) as pp
-                        from raw_session_replay_events
-                        where min_first_timestamp <= now()
-                        and min_first_timestamp >= now() - toIntervalDay(7)
-                        group by session_id
-                        having date_diff('second', min (min_first_timestamp), max (max_last_timestamp)) > 5000
-                        ) as q
-                    group by p
-                    order by c desc
-                        limit 10
+                  WITH
+session_distinct_ids AS (
+    SELECT any(distinct_id) AS di
+    FROM raw_session_replay_events
+    WHERE min_first_timestamp <= now()
+      AND min_first_timestamp >= now() - toIntervalDay(7)
+    GROUP BY session_id
+    HAVING date_diff('second', min(min_first_timestamp), max(max_last_timestamp)) > 5
+),
+
+person_ids as (
+  SELECT
+    p.id AS person_id, p.properties as person_properties, p.pdi.distinct_id as the_pdi
+  FROM persons p
+  where p.pdi.distinct_id in (select di from session_distinct_ids)
+),
+
+counted_sessions AS (
+    SELECT di AS sess_di, count() AS c
+    FROM (
+        SELECT any(distinct_id) AS di
+        FROM raw_session_replay_events
+        WHERE min_first_timestamp <= now()
+          AND min_first_timestamp >= now() - toIntervalDay(7)
+          AND distinct_id IN (SELECT di FROM session_distinct_ids)
```

Comment:
AFAIK, this appearing twice means it'll be calculated twice; clickhouse CTEs are not precalculated or anything like this, it just inlines the SQL. That might be ok if session_distinct_ids is cheap (it doesn't look too bad)

Reply:
I'm not sure it'd help you, we don't do person squashing there either.

We experimented with argMax(person_id, timestamp) to get the final person id for a session (as an approx person ID count, without needing to do any joins), but it was a bit too inaccurate on real-world data.